GitUI/Editor/FileViewer.cs

```diff
@@ -575,26 +576,32 @@ string GetFileTextIfBlobExists()
         public Task ViewFileAsync(string fileName, bool isSubmodule = false, Action? openWithDifftool = null)
         {
             string? fullPath = _fullPathResolver.Resolve(fileName);
-
-            if (isSubmodule && !GitModule.IsValidGitWorkingDir(fullPath))
-            {
-                return ViewTextAsync(fileName, "Invalid submodule: " + fileName);
-            }
-
             Validates.NotNull(fullPath);
+            Debug.Assert(Path.IsPathFullyQualified(fullPath), "Path must be resolved and fully qualified");
```

Comment:
Is `fullPath` really not `null` here now? Then its type could be (non-nullable) `string`.

Reply:
_fullPathResolver.Resolve() is string?, but the result should not be null in our case (and null was actually OK).
ValidatesNotNull() must check the result, to follow the contract of other methods.